projects/plugins/jetpack/extensions/blocks/payment-buttons/payment-buttons.php

```diff
@@ -28,7 +28,18 @@ function register_block() {
 	if ( \Jetpack_Memberships::is_enabled_jetpack_recurring_payments() ) {
 		Blocks::jetpack_register_block(
 			BLOCK_NAME,
-			array( 'render_callback' => __NAMESPACE__ . '\render_block' )
+			array(
+				'render_callback' => __NAMESPACE__ . '\render_block',
+				'supports'        => array(
```

Comment:
Yes, because that's how some of the css works in Gutenberg at view time. It uses the [`gutenberg_render_layout_support_flag` PHP function ](https://github.com/WordPress/gutenberg/blob/0b30a4cb34d39c9627b6a3795a18aee21019ce25/lib/block-supports/layout.php#L294) PHP function to register some styles for a `wp-container-N` class and updates the output block to include that classname. Without support declaration that function exits early.

Absence of this for people without a valid wpcom plan causes the [display to be incorrect](https://github.com/Automattic/jetpack/pull/26134#issuecomment-1249424791)

Reply:
I wonder how the Core Buttons block works then, since it doesn't declare any `__experimentalLayout` support in the backend. Perhaps [the block.json file](https://github.com/WordPress/gutenberg/blob/0b30a4cb34d39c9627b6a3795a18aee21019ce25/packages/block-library/src/buttons/block.json#L34) is used by both the frontend and the backend?

Reply:
That's exactly what happens, moving the JP blocks to use block.json seemed a little too far out of scope for this PR, maybe worth doing another time though.